### DISCUSSION TITLE
Is there a way to get a forecast for multiple gridpoints at the same time?

### DISCUSSION THREAD

**Discussion by iStarke** [weather-gov/api#429] [Q&A]
I'm interested in getting the detailed forecast such as the one you get from [https://api.weather.gov/gridpoints/LWX/96,70](url) but for a larger area. It would be unwanted and unefficient to loop over a set of gridpoints.

Without knowing what the rate limit for the API is, I think someone would be very upset if I tried to make one request for every gridpoint in the entire country.

Is this possible?

**Comment by scadergit** [comment]
That feature is not available at this time. It has been asked enough to warrant discussion for the roadmap, but no dates on when that might be implemented.

**Reply by sameersheikh22** [reply]
Hi @scadergit 
Any update regarding this feature?

**Reply by sullynole** [reply]
Since the API is designed around point forecasts, an alternative option is NDFD grib data if you want forecast data for the entire US: https://vlab.noaa.gov/web/mdl/ndfd

**Reply by sameersheikh22** [reply]
I need current observed temperature data for multiple zip codes at a time. There are around 40k zip codes in USA. 40k api calls taking lot of time!
